app/assets/src/components/ui/controls/dropdowns/ThresholdFilterDropdown.jsx

```diff
@@ -23,12 +23,42 @@ class ThresholdFilterDropdown extends React.Component {
     let newThresholds = props.thresholds.filter(
       ThresholdFilterDropdown.isThresholdValid
     );
-    if (newThresholds !== state.thresholds) {
-      return { thresholds: newThresholds };
+    if (
+      !ThresholdFilterDropdown.areThresholdsFiltersEqual(
+        newThresholds,
+        state.oldThresholds
+      )
+    ) {
+      return {
+        thresholds: newThresholds,
+        oldThresholds: newThresholds
+      };
     }
     return null;
   }
 
+  static areThresholdsFiltersEqual(tfs1, tfs2) {
```

Comment:
should this go into some helper library for deep comparison of objects?

Reply:
This one is very specific of this object... so I would say no. A full object comparison is more complicated.